### PR TITLE
MCQ checkboxes not selectable on iOS devices

### PR DESCRIPTION
**Pull request by eleanor-heath**
Fixes: https://github.com/adaptlearning/adapt-contrib-mcq/issues/203

Testing notes
**Description**
MCQ/GMCQ item checkboxes not able to be selected on iOS devices.

**Testing steps**

- Add a course with a MCQ component
- Add a few answers to the component
- Set _Selectable to 2
- Preview on iOS device

**Expected results**
I should be able to select checkboxes on my iOS device.

**Actual results**
I am not able to select checkboxes on my iOS device.

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by swashbuck — APPROVED** [review]
👍 

Fixed. Tested on iPhone SE iOS 16.3.1

File: less/mcq.less [hunk 1]
```diff
@@ -53,8 +44,12 @@
     display: none;
   }
 
+  &__input {
+    display: none;
```
Comment (oliverfoster) [inline less/mcq.less:48]: I can't remember why we had opacity rather than display: none. will check with screen readers etc to make sure we're all good.

Comment (oliverfoster) [reply less/mcq.less:48]: It's because you can't tab onto the inputs if it's display: none

**Comment by oliverfoster** [comment]
@eleanor-heath can you give that another try please? It seems that nesting the input in the label is sufficient to fix the ios issue.

To have keyboard and screen reader accessibility the input needs to be focusable in the DOM. To make sure the label reads correctly I have moved the aria label, with all the correctness strings, onto the label from the input. I've hidden all of the text inside the label from the screen reader such that only the aria label is read and not also the label text.

**Comment by oliverfoster** [comment]
Yea, that was entirely painful. After debugging properly on ios and macos, focus and blur were changing the jsx enough to create a new input each time it was focused, stopping the click event from firing. I've prevented focus and blur handlers on touch devices.

All screen reader/os/browser combinations and tabbing are working correctly again.

**Comment by eleanor-heath** [comment]
> Yea, that was entirely painful. After debugging properly on ios and macos, focus and blur were changing the jsx enough to create a new input each time it was focused, stopping the click event from firing. I've prevented focus and blur handlers on touch devices.
> 
> All screen reader/os/browser combinations and tabbing are working correctly again.

thank you so much

**Review by eleanor-heath — COMMENTED** [review]
Cannot approve a PR I have open but this Im writing this to say Ive approved Olivers recent changes - works on iOS device for both MCQ selectable checkboxes and radio options ✅

**Review by oliverfoster — APPROVED** [review]

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by StuartNicholls — APPROVED** [review]

**Pull request merged**